### PR TITLE
language select and translations

### LINKED ISSUE #853: Translation and Language Selection
### Describe the feature

The landing page should be translated to German and provide a dropdown for changing the language

### Why is this feature needed?

Because not all visitors might be happy with English texts

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by florian-sabonchi**
## Which issues does this pull request close?
closes #853

### CODE REVIEW

File: landing-page/components/Footer.tsx [hunk 1]
```diff
@@ -1,13 +1,36 @@
...
 import pluginConfig from '../utils/CookieConsentConfig'
 import FooterLinkGroup from './FooterLinkGroup'
-
 import 'vanilla-cookieconsent/dist/cookieconsent.css'
 
+type FooterTranslation = {
+    Socials: string,
+    General: string,
+    Products: string,
+    Development: string
```
Comment (MaxSchaefer) [inline landing-page/components/Footer.tsx:16]:
```suggestion
    socials: string,
    general: string,
    products: string,
    development: string
```
Lower case

File: landing-page/components/Footer.tsx [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+    Socials: 'Socials',
+    General: 'General',
+    Products: 'Products',
+    Development: 'Development',
+  },
+  de: {
+    Socials: 'Social media',
+    General: 'Allgemein',
+    Products: 'Produkte',
+    Development: 'Entwicklung',
```
Comment (MaxSchaefer) [inline landing-page/components/Footer.tsx:30]:
```suggestion
    Socials: 'social',
    General: 'allgemein',
    Products: 'produkte',
    Development: 'entwicklung',
```

File: landing-page/components/Header.tsx [hunk 3]
```diff
@@ -46,13 +50,32 @@ const items: SubLinkType[] = [
...
+    team: 'team',
+    talks: 'talks',
+    contact: 'contact',
+  },
+  de: {
+    products: 'Produkte',
+    story: 'Geschichte',
+    team: 'Team',
+    talks: 'tasks',
+    contact: 'Kontakt',
```
Comment (MaxSchaefer) [inline landing-page/components/Header.tsx:66]:
```suggestion
    products: 'produkte',
    story: 'geschichte',
    team: 'team',
    talks: 'tasks',
    contact: 'kontakt',
```

File: landing-page/components/sections/landing/ExpansionSection.tsx [hunk 4]
```diff
@@ -1,21 +1,46 @@
...
+  en: {
+    germanyHealthcareSystem: 'Germany\'s Healthcare System',
+    hospitals: 'hospitals',
+    healthcareWorkers: 'healthcare workers',
+    gdp: 'of GDP'
+  },
+  de: {
+    germanyHealthcareSystem: 'Deutsches Gesundheitssystem',
+    hospitals: 'Krankenhäuser',
+    healthcareWorkers: 'Gesundheitspersonal',
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/ExpansionSection.tsx:22]:
```suggestion
    healthcareWorkers: 'healthcare workers',
```

File: landing-page/components/sections/landing/StartSection.tsx [hunk 5]
```diff
@@ -1,7 +1,26 @@
...
+  discoverOurVision: string
+}
+
+const defaultLandingPageTranslation = {
+  en: {
+    tryTheDemo: 'Try the demo!',
+    discoverOurVision: 'Discover Our Vision'
+  },
+  de: {
+    tryTheDemo: 'Probieren Sie die Demo!',
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/StartSection.tsx:17]:
```suggestion
    tryTheDemo: 'Teste die Demo!',
```

File: landing-page/components/sections/landing/StartSection.tsx [hunk 6]
```diff
@@ -1,7 +1,26 @@
...
+    discoverOurVision: 'Entdecken Sie unsere Vision'
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/StartSection.tsx:18]:
```suggestion
    discoverOurVision: 'Entdecken unsere Vision'
```

File: landing-page/components/sections/landing/Story.tsx [hunk 7]
```diff
@@ -1,11 +1,51 @@
...
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+import type { Languages } from '@helpwave/common/hooks/useLanguage'
 
 type StoryBlockProps = {
   pill?: string,
   header: string,
   content: string
 }
 
+type StoryTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/Story.tsx:12]: These translations should not be part of this pull request.

File: landing-page/components/sections/story/StartSection.tsx [hunk 8]
```diff
@@ -1,47 +1,84 @@
 import { tw } from '@helpwave/common/twind'
 import { Popcorn } from 'lucide-react'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
 
-const StartSection = () => {
+type StoryTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/story/StartSection.tsx:6]: These translations should not be part of this pull request.

File: landing-page/components/sections/talks/EpisodeSection.tsx [hunk 9]
```diff
@@ -4,6 +4,21 @@ import Link from 'next/link'
...
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+
+type EpisodeTranslation = {
+  allEpisodes: string
+}
+
+const defaultEpisodeTranslation = {
+  en: {
+    allEpisodes: 'All the episodes...'
```
Comment (MaxSchaefer) [inline landing-page/components/sections/talks/EpisodeSection.tsx:16]:
```suggestion
    allEpisodes: 'All episodes...'
```

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: landing-page/components/Footer.tsx [hunk 10]
```diff
@@ -58,7 +58,7 @@ const linkGroups = [
       { name: 'LOI', link: 'https://cdn.helpwave.de/helpwave_letter_of_intent.pdf' },
       { name: 'Tech-Radar', link: '/tech-radar', openInCurrentTab: true },
     ],
-    Products: [
```
Comment (use-to) [inline landing-page/components/Footer.tsx:38]: Pretty sure this uppercase was on purpose, as the footer uses the key as input.

**Review by use-to — CHANGES_REQUESTED** [review]

File: landing-page/components/Footer.tsx (hunk 10 above)
Comment (florian-sabonchi) [reply landing-page/components/Footer.tsx:38]: Okay, i need to check again the Problem ist that i am using the Name AS translation key, so the orthography is importan

File: landing-page/components/Footer.tsx [hunk 11]
```diff
@@ -1,13 +1,36 @@
...
+    products: string,
+    development: string
+}
+
+const defaultFooterTranslation = {
+  en: {
+    socials: 'Socials',
+    general: 'General',
+    products: 'Products',
+    development: 'Development',
```
Comment (MaxSchaefer) [inline landing-page/components/Footer.tsx:24]:
```suggestion
    socials: 'socials',
    general: 'general',
    products: 'products',
    development: 'development',
```

File: landing-page/components/Footer.tsx [hunk 12]
```diff
@@ -68,7 +92,7 @@ const Footer = () => {
         {linkGroups.map((group, index) => (
           <div key={index} className={tw('mobile:w-full desktop:w-[192px] mobile:text-center')}>
             {Object.entries(group).map(([title, links]) => (
-              <FooterLinkGroup key={title} title={title} links={links} />
+              <FooterLinkGroup key={title} title={translation[title.toLowerCase() as keyof typeof translation] } links={links} />
```
Comment (MaxSchaefer) [inline landing-page/components/Footer.tsx:95]: Please add a Todo here. The technical `.toLowerCase()` on the key is weird. There must be an alternative ...

File: landing-page/components/Footer.tsx [hunk 13]
```diff
@@ -1,13 +1,36 @@
...
 import 'vanilla-cookieconsent/dist/cookieconsent.css'
 
+type FooterTranslation = {
+    socials: string,
+    general: string,
+    products: string,
+    development: string
+}
+
+const defaultFooterTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/Footer.tsx:19]:
```suggestion
const defaultFooterTranslation: Record<Languages, FooterTranslation> = {
```

File: landing-page/components/Header.tsx [hunk 14]
```diff
@@ -46,13 +50,32 @@ const items: SubLinkType[] = [
   },
 ]
 
+const defaultHeaderTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/Header.tsx:53]: Type `HeaderTranslation` are missing entirely
```suggestion
const defaultHeaderTranslation: Record<Languages, HeaderTranslation> = {
```

File: landing-page/components/Header.tsx [hunk 15]
```diff
@@ -110,6 +133,14 @@ const Header = () => {
               <MenuIcon size={32}/>
             </button>
           </div>
+          <Select
+            value={language}
+            onChange={(language: string) => setLanguage(language as Languages)}
+            options={[
+              { value: 'de', label: 'Deutsch' },
+              { value: 'en', label: 'Englisch' }
```
Comment (MaxSchaefer) [inline landing-page/components/Header.tsx:141]:
```suggestion
              { value: 'de', label: 'Deutsch' },
              { value: 'en', label: 'English' }
```

File: landing-page/components/sections/landing/ExpansionSection.tsx [hunk 16]
```diff
@@ -1,21 +1,46 @@
...
 
-const ExpansionSection = () => {
+type ExpansionTranslation = {
+  germanyHealthcareSystem: string,
+  hospitals: string,
+  healthcareWorkers: string,
+  gdp: string
+}
+
+const defaultExpansionTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/ExpansionSection.tsx:12]:
```suggestion
const defaultExpansionTranslation: Record<Languages, ExpansionTranslation> = {
```

File: landing-page/components/sections/talks/EpisodeSection.tsx [hunk 17]
```diff
@@ -4,6 +4,21 @@ import Link from 'next/link'
...
 import { LoadingAnimation } from '@helpwave/common/components/LoadingAnimation'
 import { useQuery } from '@tanstack/react-query'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+
+type EpisodeTranslation = {
+  allEpisodes: string
+}
+
+const defaultEpisodeTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/talks/EpisodeSection.tsx:14]:
```suggestion
const defaultEpisodeTranslation: Record<Languages, EpisodeTranslation> = {
```

File: landing-page/components/sections/talks/StartSection.tsx [hunk 18]
```diff
@@ -2,8 +2,25 @@ import { tw, tx } from '@helpwave/common/twind'
...
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
 
-const StartSection = () => {
+type taskTranslation = {
+  fusionOfHealthcareAndComputerScience: string
+
+}
+
+const defaultTaskTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/talks/StartSection.tsx:13]:
```suggestion
type TaskTranslation = {
  fusionOfHealthcareAndComputerScience: string

}

const defaultTaskTranslation: Record<Languages, TaskTranslation> = {
```

File: landing-page/components/sections/landing/StartSection.tsx [hunk 19]
```diff
@@ -1,7 +1,26 @@
...
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
 
-const StartSection = () => {
+type landingPageTranslation = {
+  tryTheDemo: string,
+  discoverOurVision: string
+}
+
+const defaultLandingPageTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/StartSection.tsx:11]:
```suggestion
type LandingPageTranslation = {
  tryTheDemo: string,
  discoverOurVision: string
}

const defaultLandingPageTranslation: Record<Languages, LandingPageTranslation> = {
```

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: landing-page/components/sections/talks/EpisodeSection.tsx [hunk 20]
```diff
@@ -4,6 +4,22 @@ import Link from 'next/link'
 
 import { LoadingAnimation } from '@helpwave/common/components/LoadingAnimation'
 import { useQuery } from '@tanstack/react-query'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+import type { Languages } from '@helpwave/common/hooks/useLanguage'
+
+type EpisodeTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/talks/EpisodeSection.tsx:11]:
```suggestion
type EpisodeSectionTranslation = {
```
Check also the other mentions of `Episode...`

File: landing-page/components/sections/talks/StartSection.tsx [hunk 21]
```diff
@@ -2,8 +2,26 @@ import { tw, tx } from '@helpwave/common/twind'
 import { Link as LinkIcon } from 'lucide-react'
 import Image from 'next/image'
 import Link from 'next/link'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+import type { Languages } from '@helpwave/common/hooks/useLanguage'
 
-const StartSection = () => {
+type TaskTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/talks/StartSection.tsx:9]:
```suggestion
type TalksTranslation = {
```

File: landing-page/components/sections/landing/ExpansionSection.tsx [hunk 22]
```diff
@@ -1,21 +1,47 @@
 import { tw } from '@helpwave/common/twind'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+import type { Languages } from '@helpwave/common/hooks/useLanguage'
 
-const ExpansionSection = () => {
+type ExpansionTranslation = {
```
Comment (MaxSchaefer) [inline landing-page/components/sections/landing/ExpansionSection.tsx:6]:
```suggestion
type ExpansionSectionTranslation = {
```

**Review by MaxSchaefer — APPROVED** [review]

**Comment by MaxSchaefer** [comment]
@use-to requested changes

**Comment by use-to** [comment]
> @use-to requested changes

As you can see above. The thread should not have been resolved. Still requested changes.

**Comment by use-to** [comment]
Let's also think about spelling the menu items uppercase like the current footer, too!

**Review by use-to — CHANGES_REQUESTED** [review]
Let's put the language switcher into the footer. Also reenable the language auto-detect feature introduced by @JannikWibker before.

File: lib/components/user-input/Select.tsx [hunk 23]
```diff
@@ -92,7 +97,7 @@ export const Select = <T, >({
                 <Menu.Item key={`item${index}`}>
                   {
                     <div
-                      className={tx('px-4 py-2 overflow-hidden whitespace-nowrap text-ellipsis border-2 border-t-0',
+                      className={tx(' px-4 py-2 overflow-hidden whitespace-nowrap text-ellipsis border-2 border-t-0',
```
Comment (use-to) [inline lib/components/user-input/Select.tsx:100]:
```suggestion
                      className={tx('px-4 py-2 overflow-hidden whitespace-nowrap text-ellipsis border-2 border-t-0',
```

File: landing-page/components/Footer.tsx [hunk 24]
```diff
@@ -1,13 +1,38 @@
...
+    general: 'general',
+    products: 'products',
+    development: 'development',
+  },
+  de: {
+    socials: 'social',
+    general: 'allgemein',
+    products: 'produkte',
+    development: 'entwicklung',
+  }
```
Comment (use-to) [inline landing-page/components/Footer.tsx:33]: Make the titles uppercase.

**Review by use-to — CHANGES_REQUESTED** [review]

File: landing-page/components/Footer.tsx (hunk 24 above)
Comment (MaxSchaefer) [reply landing-page/components/Footer.tsx:33]: nit: Why should we differ capitalization between the navigation bar titles and footer titles? The current state of lowercase navbar and uppercase footer seams a little bit inconsistent. I would recommend all lowercase in all languages.

Comment (use-to) [reply landing-page/components/Footer.tsx:33]: In order to create a visual difference of the footer titles (besides bold). Our current state is uppercase and it looks really appealing. We can discuss this on another issues. To me, this issues is not about redesigning - it's about making the current state translated.

**Review by use-to — APPROVED** [review]
@florian-sabonchi let's finish this pr at this point... Good job 🥳

**Comment by use-to** [comment]
Please face the following problems in separate issues:
image
